src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -274,6 +278,22 @@ internal void WorkerInitResponse(GrpcEvent initEvent)
             _workerInitTask.SetResult(true);
         }
 
+        private void ParseAndLogWorkerMetadata()
+        {
+            if (_initMessage?.WorkerMetadata == null)
+            {
+                return;
+            }
+
+            _initMessage.WorkerMetadata.RuntimeName = string.IsNullOrEmpty(_initMessage.WorkerMetadata.RuntimeName)
+                                                        ? _workerConfig.Description.Language : _initMessage.WorkerMetadata.RuntimeName;
+            _initMessage.WorkerMetadata.RuntimeVersion = string.IsNullOrEmpty(_initMessage.WorkerMetadata.RuntimeVersion)
```

Comment:
So, we are falling back to workerConfig values if worker GRPC message does not have certain properties. It may be possible for us to log a mix of what was sent from worker and what is present in workerconfig., Would that cause any confusion? Should we be explicit that we are using the fallback version value?

Reply:
I don't think it would cause confusion given these two values are intentionally optional, and this is just for telemetry so there is not code impact. I can add a log here to indicate we are using the worker config but I don't feel it is necessary

Reply:
Makes sense. Agree that it may be uncessary.